fem/tmop.cpp

```diff
@@ -14,12 +14,470 @@
 #include "pgridfunc.hpp"
 #include "tmop_tools.hpp"
 #include "../general/forall.hpp"
+#include "../linalg/dual.hpp"
 
 namespace mfem
 {
 
+/* AD related definitions below ========================================*/
+
+/// MFEM native AD-type for first derivatives
+typedef internal::dual<real_t, real_t> ADFType;
+/// MFEM native AD-type for second derivatives
+typedef internal::dual<ADFType, ADFType> ADSType;
```

Comment:
How about `AD1Type` and `AD2Type`? Otherwise `ADS` might be confused with something else.